core/src/main/java/com/linecorp/armeria/server/Http1ServerKeepAliveHandler.java

```diff
@@ -27,9 +27,9 @@ final class Http1ServerKeepAliveHandler extends Http1KeepAliveHandler {
 
     Http1ServerKeepAliveHandler(Channel channel, Timer keepAliveTimer,
                                 long idleTimeoutMillis, long maxConnectionAgeMillis,
-                                int maxNumRequestsPerConnection) {
+                                int maxNumRequestsPerConnection, boolean keepAliveOnPing) {
         super(channel, "server", keepAliveTimer, idleTimeoutMillis, /* pingIntervalMillis(unsupported) */ 0,
-              maxConnectionAgeMillis, maxNumRequestsPerConnection);
+              maxConnectionAgeMillis, maxNumRequestsPerConnection, keepAliveOnPing);
```

Comment:
HTTP/1 server can't send a PING or distinguish a PING request.
```suggestion
              maxConnectionAgeMillis, maxNumRequestsPerConnection, false);
```

Reply:
Hmm, I thought we would have this option because considering `OPTIONS` as ping but I think I misunderstood something. https://github.com/line/armeria/pull/4806#discussion_r1158436288

Based on this convo I thought HTTP1.1 assumes OPTIONS as keep alive?

Reply:
> Based on this convo I thought HTTP1.1 assumes OPTIONS as keep alive?

That is correct. I guess what @ikhoon is trying to say is that we only need to consider it on the client-side.
On the server-side, When HTTP/1.1 is used, we don't distinguish the ping request.
When the server receives a ping request, the `lastConnectionIdleTime` is reset anyway:
https://github.com/line/armeria/blob/6d50c358c99865b463a0f772f04d3e13a8bc50e1/core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java#L131

We can also override `onPing` method in `Http1ServerKeepAliveHandler` such as we do for `writePing` to make it clearer.

Reply:
We can only apply `keepAliveOnPing` to the client side for HTTP/1.

Reply:
Got it, I got confused and applied it to the server instead of the client.

Do I need to override that method in this PR? A helping hand can help 😓

Reply:
Oh, you don't have to. 😉